Flickr/Flickr/ImageCollectionViewController.swift

```diff
@@ -124,36 +260,147 @@ class ImageCollectionViewController: UIViewController, UICollectionViewDelegate,
+            let cellPopular = collectionView.dequeueReusableCell(withReuseIdentifier: "PopularImage Cell",
+                                                                 for: indexPath)
+            if let imageCell = cellPopular as? ImageCollectionViewCell {
+                guard let url = popularImageData[indexPath.row].url else {
+                    return cellPopular
+                }
+                imageCell.fetchImage(url: url)
+                return cellPopular
             }
-            imageCell.fetchImage(url: gettedUrl)
         }
-        return cell
+        return UICollectionViewCell()
     }
     
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {    // in process
-        if segue.identifier == Constants.SegueIdentifier.GallerySegue {
-            if let gvcvc = segue.destination as? GalleryViewingCollectionViewController {
-                gvcvc.photoGalleryData = imageData
-                if let cell = sender as? ImageCollectionViewCell,
-                    let tweetMedia = gvcvc.photoGalleryData {
-
-                }
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        if isSearching == false {
+            return popularImageSizes[indexPath.row]
+        } else {
+            return searchImageSizes[indexPath.row]
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        return ConstantNumbers.basicIndent
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
+        return ConstantNumbers.basicIndent/4
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, willDisplay cell: UICollectionViewCell, forItemAt indexPath: IndexPath) {
+        if collectionView == self.popularCollectionView {
+            if indexPath.row == (pageFlickr * ConstantNumbers.perPage) - ConstantNumbers.lastCells {
+                pageFlickr += 1
+                performFlickrPopular()
+            }
+        }
+        if collectionView == self.searchCollectionView {
+            if indexPath.row == (pageFlickr * ConstantNumbers.perPage) - ConstantNumbers.lastCells {
+                pageFlickr += 1
+                performTextSearch()
             }
         }
     }
-
+    
+    func scrollViewWillBeginDragging(_ scrollView: UIScrollView) {
+        recentTable_GoHide()
+    }
+    
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        let actualPosition = scrollView.panGestureRecognizer.translation(in: scrollView.superview)
+        if actualPosition.y > 0 {
+            self.view.layoutIfNeeded()                              //If u want animated cells while scrolling comment this
```

Comment:
you probably could call layoutIfNeeded not for entire self.view but only for searchView so it will not triger table animation